### PR TITLE
EOS-16248 User interface for cluster start CLI

### PR DESCRIPTION
**Pull request by shindeam**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-16248 User interface for cluster start CLI
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  No
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Implement user interface for cluster start CLI
  </code>
</pre>
## Solution
<pre>
  <code>
    - Code implementation in cluster manager to point to the cluster controller
    - Code implementation in CortxClusterManager to process the request provided at CLI level
    - Set output to cli of our actual core library function response
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
     -bash-4.2$ sudo cortxha cluster start --all --json
     "{\"status\": \"InProgress\", \"msg\": \"Cluster start operation performed\"}"

     -bash-4.2$ sudo cortxha cluster start --server --json
     "{\"status\": \"InProgress\", \"msg\": \"Cluster start operation performed\"}"
  </code>
</pre>

### CODE REVIEW

File: ha/cli/exec/clusterExecutor.py [hunk 1]
```diff
@@ -15,118 +15,34 @@
...
-         Standby with resource(s) running:
-         Maintenance:
-         Offline:
-        Pacemaker Remote Nodes:
-         Online:
-         Standby:
-         Standby with resource(s) running:
-         Maintenance:
-         Offline:
+        Validate command for cluster start executor
```
Comment (ajaysrivas) [inline ha/cli/exec/clusterExecutor.py:34]: This must be implemented and should return True/False.
Please check "add node" command implementation.

File: ha/cli/exec/clusterExecutor.py [hunk 2]
```diff
@@ -15,118 +15,34 @@
...
-            if self.active_nodes == False:
-                # wait for 5 seconds and retry
-                time.sleep(5)
-                self.get_nodes_status()
-                if self.active_nodes == False:
-                    Log.info("Cluster started; nodes not online")
-                    raise HAClusterStart("Cluster started; nodes not online")
-
-        Log.info("Cluster started successfully")
+        result = self._controllers[const.CLUSTER_CONTROLLER].start()
```
Comment (ajaysrivas) [inline ha/cli/exec/clusterExecutor.py:43]: No code here. It should have all the logic implemented earlier.

**Review by ajaysrivas — CHANGES_REQUESTED** [review]

File: ha/cli/exec/commandExecutor.py [hunk 3]
```diff
@@ -22,11 +22,25 @@
...
+from cortx.utils.conf_store.conf_store import Conf
 from ha.core.error import HAInvalidPermission
+from ha.core.config.config_manager import ConfigManager
+from ha.core.controllers.element_controller_factory import ElementControllerFactory
+
 
 class CommandExecutor:
     def __init__(self):
         self._is_hauser = False
+        self._cluster_type = Conf.get(const.HA_GLOBAL_INDEX, "CLUSTER_MANAGER.cluster_type")
```
Comment (ajaysrivas) [inline ha/cli/exec/commandExecutor.py:34]: Why this code change?

File: ha/cli/exec/commandExecutor.py [hunk 4]
```diff
@@ -22,11 +22,25 @@
...
+        self._env = Conf.get(const.HA_GLOBAL_INDEX, "CLUSTER_MANAGER.env")
+        ConfigManager.load_controller_schema()
+        self._controllers = ElementControllerFactory.init_controller(self._env, self._cluster_type)
+        for controller in self._controllers.keys():
+            Log.info(f"Adding {controller} property to cluster manager.")
+            # Add property method for controller
+            # Example: cm.cluster_controller.start()
+            # Find more example in test case.
+            self.__dict__[controller] = self._controllers[controller]
```
Comment (ajay-paratmandali) [inline ha/cli/exec/commandExecutor.py:43]: env and cluster type already handled in cluster manager

this not needed
only 
cm.cluster_controller.start()
should work

File: ha/cli/exec/clusterExecutor.py (hunk 1 above)
Comment (shindeam) [reply ha/cli/exec/clusterExecutor.py]: Updated code changes.

File: ha/cli/exec/clusterExecutor.py (hunk 2 above)
Comment (shindeam) [reply ha/cli/exec/clusterExecutor.py]: Added logic

File: ha/cli/exec/commandExecutor.py (hunk 4 above)
Comment (shindeam) [reply ha/cli/exec/commandExecutor.py]: Yes did same :) thank you..

**Review by ajaysrivas — APPROVED** [review]

**Review by ajay-paratmandali — APPROVED** [review]

**Review by indrajitzagade — APPROVED** [review]

**Pull request merged**